### PR TITLE
[Merged by Bors] - Implement resource requests and limits for OPA pods

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- added cpu and memory resource limits

fixes https://github.com/stackabletech/opa-operator/issues/325

This currently only sets resource limits for the `opa` container. I created an issue https://github.com/stackabletech/opa-operator/issues/348 in case we ever need to provide the `opa-bundle-builder` container with more resources.

Test:
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/opa-operator-it-custom/12/

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,16 +4,22 @@ All notable changes to this project will be documented in this file.
...
 
 - Include chart name when installing with a custom release name ([#313], [#314]).
 - `operator-rs` `0.15.0` -> `0.22.0` ([#315]).
 
-[#313]: https://github.com/stackabletech/trino-operator/pull/313
-[#314]: https://github.com/stackabletech/trino-operator/pull/314
-[#315]: https://github.com/stackabletech/trino-operator/pull/315
+[#313]: https://github.com/stackabletech/opa-operator/pull/313
+[#314]: https://github.com/stackabletech/opa-operator/pull/314
+[#315]: https://github.com/stackabletech/opa-operator/pull/315
```
Comment (sbernauer) [inline CHANGELOG.md:22]:
```suggestion
[#315]: https://github.com/stackabletech/opa-operator/pull/315
[#347]: https://github.com/stackabletech/opa-operator/pull/347
```

File: docs/modules/ROOT/pages/usage.adoc [hunk 2]
```diff
@@ -78,3 +78,67 @@ servers:
...
+    default:
+      config:
+        resources:
+          memory:
+            limit: '2Gi'
+----
+
+In this example, each OPA container in the `default` role group will have a maximum of 2 gigabytes of memory. To be more precise, these memory limits apply to the container running OPA but not to any sidecar containers (e.g. the BundleBuilder) that are part of the pod.
+
+A general rule of thumb on how to size the memory for OPA is described https://www.openpolicyagent.org/docs/latest/policy-performance/#resource-utilization[here].
```
Comment (sbernauer) [inline docs/modules/ROOT/pages/usage.adoc:103]:
```suggestion
A general rule of thumb on how to size the memory for OPA is described https://www.openpolicyagent.org/docs/latest/policy-performance/#resource-utilization[in the opa docs].
```

File: docs/modules/ROOT/pages/usage.adoc [hunk 3]
```diff
@@ -78,3 +78,67 @@ servers:
...
+  roleGroups:
+    default:
+      config:
+        resources:
+          requests:
+            cpu: 200m
+            memory: 2Gi
+          limits:
+            cpu: "2"
+            memory: 2Gi
```
Comment (sbernauer) [inline docs/modules/ROOT/pages/usage.adoc:139]: We keep making this error. There must be a broken copy/paste somewhere else ;)
```suggestion
          cpu:
            min: '200m'
            max: "4"
          memory:
            limit: '2Gi'
```

File: tests/templates/kuttl/resources/01-install-opa.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: opa.stackable.tech/v1alpha1
+kind: OpaCluster
+metadata:
+  name: opa
+spec:
+  version: {{ test_scenario['values']['opa'] }}
```
Comment (sbernauer) [inline tests/templates/kuttl/resources/01-install-opa.yaml.j2:7]: What do you think about `opa-latest`? ;)

File: CHANGELOG.md (hunk 1 above)
Comment (maltesander) [reply CHANGELOG.md:22]: I dont quite get this? This PR (347) is under unreleased.

File: docs/modules/ROOT/pages/usage.adoc (hunk 2 above)
Comment (maltesander) [reply docs/modules/ROOT/pages/usage.adoc]: Thanks done

File: docs/modules/ROOT/pages/usage.adoc (hunk 3 above)
Comment (maltesander) [reply docs/modules/ROOT/pages/usage.adoc]: Thanks. Yeah that makes me wonder where else this is still in.

File: tests/templates/kuttl/resources/01-install-opa.yaml.j2 (hunk 4 above)
Comment (maltesander) [reply tests/templates/kuttl/resources/01-install-opa.yaml.j2:7]: OPA is so lightweight, i figured here it does not hurt to have tested all the versions. Although we are testing nothing in the OPA pods (rather if its set in the daemonset correctly), so i could do it. What do you think?

File: CHANGELOG.md (hunk 1 above)
Comment (sbernauer) [reply CHANGELOG.md:22]: Oh I'm stupid, my fault

File: tests/templates/kuttl/resources/01-install-opa.yaml.j2 (hunk 4 above)
Comment (sbernauer) [reply tests/templates/kuttl/resources/01-install-opa.yaml.j2:7]: OPA images are small agreed. On the other hand i would prefer consistency across the operators.
Feel free too leave it as is but than please put in a comment like "Testing all OPA versions here as the images and containers are very lightweight"

**Review by sbernauer — APPROVED** [review]
LGTM % opa-latest discussion

File: tests/templates/kuttl/resources/01-install-opa.yaml.j2 (hunk 4 above)
Comment (maltesander) [reply tests/templates/kuttl/resources/01-install-opa.yaml.j2]: done https://github.com/stackabletech/opa-operator/pull/347/commits/1c6f0593d758c7d3995b5391e1aaafd39dee235e

**Review by sbernauer — APPROVED** [review]
thx!

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**